gnocchi/rest/influxdb.py

```diff
@@ -0,0 +1,316 @@
+
+                for field_name, field_value in six.iteritems(fields):
+                    if isinstance(field_value, str):
+                        # We do not support field value that are not numerical
+                        continue
+
+                    # Metric name is the:
+                    # <measurement>.<field_key>@<tag_key>=<tag_value>,…
+                    # with tag ordered
+                    # Replace "/" with "_" because Gnocchi does not support /
+                    # in metric names
+                    metric_name = (
+                        measurement + "." + field_name + tags_str
+                    ).replace("/", "_")
+
+                    resources[resource_id][metric_name].append(
+                        incoming.Measure(timestamp, field_value))
+
+            measures_to_batch = {}
+            for resource_name, metrics_and_measures in six.iteritems(
+                    resources):
+                resource_name = resource_name
+                resource_id = utils.ResourceUUID(
+                    resource_name, creator=creator)
+                LOG.debug("Getting metrics from resource `%s'", resource_name)
+                timeout = pecan.request.conf.api.operation_timeout
+                try:
+                    metrics = (
+                        self.get_or_create_resource_and_metrics.retry_with(
+                            stop=tenacity.stop_after_delay(timeout))(
+                                db, creator, resource_id, resource_name,
+                                metrics_and_measures.keys())
+                    )
+                except indexer.ResourceAlreadyExists as e:
+                    # If this function raises ResourceAlreadyExists it means
+                    # the resource might already exist as another type, we
+                    # can't continue.
+                    LOG.error("Unable to get resource `%s' for InfluxDB, "
+                              "it might already exists as another "
+                              "resource type than `%s'", resource_name, db)
+                    api.abort(400, e)
+
+                for metric in metrics:
+                    api.enforce("post measures", metric)
+
+                measures_to_batch.update(
+                    dict((metric, metrics_and_measures[metric.name])
+                         for metric in metrics
+                         if metric.name in metrics_and_measures))
+
+            LOG.debug("Add measures batch for %d metrics",
+                      len(measures_to_batch))
+            pecan.request.incoming.add_measures_batch(measures_to_batch)
+            pecan.response.status = 204
+
+            if encoding != "chunked":
+                return
+
+    @staticmethod
+    @tenacity.retry(retry=tenacity.retry_if_exception_type(
+        (indexer.NoSuchResource, indexer.NamedMetricAlreadyExists)))
+    def get_or_create_resource_and_metrics(resource_type, creator, rid,
```

Comment:
Maybe we can share it with the one in Prometheus

Reply:
Yeah I can try to do that now that Prometheus is merged.